KDEpy/utils.py

```diff
@@ -131,8 +131,75 @@ def autogrid(data, boundary_abs=3, num_points=None, boundary_rel=0.05):
     return cartesian(list_of_grids)
 
 
+def weighted_std(values, weights, ddof=0):
+    """Return the weighted standard deviation.
+
+    Examples
+    --------
+    >>> weighted_std([1, 2, 2], weights=[1, 1, 1])
+    0.4714045207910317
+    >>> weighted_std([1, 2], weights=[1, 2])
+    0.4714045207910317
+    >>> weighted_std([1, 2, 2], weights=[1, 1, 1], ddof=1)
+    0.5773502691896257
+    >>> weighted_std([1, 2], weights=[1, 2], ddof=1)
+    0.5773502691896257
+
+    """
+    values = np.asarray(values)
+    weights = np.asarray(weights)
+    assert np.all(weights > 0), "All weights must be >= 0"
+    assert isinstance(ddof, numbers.Integral), "ddof must be int"
+
+    # If the degrees of freedom is greater than zero, we need to scale results
+    if ddof > 0:
+        weights_summed = np.sum(weights)
+        factor = weights_summed / (weights_summed - ddof)
+    else:
+        factor = 1
+
+    average = np.average(values, weights=weights)
+    # Fast and numerically precise:
+    variance = np.average((values - average) ** 2, weights=weights)
+    return np.sqrt(factor * variance)
+
+
+def weighted_percentile(values, perc, weights=None):
+    """Compue the weighted percentile.
+
+    Based on: https://stackoverflow.com/a/61343915
+
+    Examples
+    --------
+    >>> weighted_percentile([1, 2, 2], 0)
+    1.0
+    >>> weighted_percentile([1, 2, 2], 1.)
+    2.0
+    >>> # These computations differ, but it does not matter
+    >>> weighted_percentile([1, 2], 0.5, [1, 2])
+    1.66666...
+    >>> weighted_percentile([1, 2, 2], 0.5)
+    2.0
+
+    """
+    values = np.asarray(values)
+
+    if weights is None:
+        weights = np.ones_like(values)
+    else:
+        weights = np.asarray(weights)
+        assert np.all(weights > 0), "All weights must be >= 0"
```

Comment:
Nit: the error message doesn't match the assertion - one is `>` and the other is `>=`